### PR TITLE
make NSC set net.ipv4.vs.conn_reuse_mode=0

### PR DESCRIPTION
**Pull request by roffe**
will fix the IPVS low throughput issues

https://github.com/kubernetes/kubernetes/pull/71114

### CODE REVIEW

File: pkg/controllers/proxy/network_services_controller.go [hunk 1]
```diff
@@ -295,6 +294,11 @@ func (nsc *NetworkServicesController) Run(healthChan chan<- *healthcheck.Control
 		return errors.New("Failed to do sysctl net.ipv4.vs.expire_quiescent_template=1 due to: %s" + err.Error())
 	}
 
+	err = ensureIpvsConnReuseMode()
+	if err != nil {
+		return errors.New("Failed to do sysctl net.ipv4.vs.conn_reuse_mode=0 due to: %s" + err.Error())
```
Comment (andrewsykim) [inline pkg/controllers/proxy/network_services_controller.go]: nit:
```go
return fmt.Errorf("failed to set net.ipv4.vs.conn_reuse_mode=0: %s", err)
```

**Comment by andrewsykim** [comment]
Do we need to check if `conn_reuse_mode` is available first? It seems like it was introduced in linux 4.1.

**Comment by roffe** [comment]
@andrewsykim glansing over the code from K8S i can't rly say they do:

https://github.com/Lion-Wei/kubernetes/blob/f39060c6047b8dce5af197a97d5f18422d93c9d1/pkg/proxy/ipvs/proxier.go

**Comment by roffe** [comment]
now we check and inform that it's a non fatal error if not there
if there is a error statting the file we also inform and continue as it's non-fatal

**Comment by murali-reddy** [comment]
thanks @roffe

File: pkg/controllers/proxy/network_services_controller.go [hunk 2]
```diff
@@ -1405,6 +1409,19 @@ func ensureIpvsConntrack() error {
...
+	sysctlPath := "/proc/sys/net/ipv4/vs/conn_reuse_mode"
+	if _, err := os.Stat(sysctlPath); err != nil {
+		if os.IsNotExist(err) {
+			glog.Infof("%s not found, skipping setting net.ipv4.vs.conn_reuse_mode=0 (non fatal error, feature introduced into kernel in 4.1)", sysctlPath)
+			return nil
+		}
+		glog.Errorf("skipping setting net.ipv4.vs.conn_reuse_mode=0, error stating: %s : %s", sysctlPath, err.Error())
+		return nil
+	}
+	return ioutil.WriteFile("sysctlPath", []byte(strconv.Itoa(0)), 0640)
```
Comment (andrewsykim) [inline pkg/controllers/proxy/network_services_controller.go]: `"sysctlPath"` -> `sysctlPath`

**Comment by roffe** [comment]
i'm fixing it with a pr

**Pull request merged**